tests/Operation/FindFunctionalTest.php

```diff
@@ -123,6 +125,45 @@ public function provideTypeMapOptionsAndExpectedDocuments()
         ];
     }
 
+    public function testMaxAwaitTimeMS()
+    {
+        $maxAwaitTimeMS = 10;
+
+        // Create a capped collection.
+        $databaseName = $this->getDatabaseName();
+        $cappedCollectionName = $this->getCollectionName() . '.capped';
+        $cappedCollectionOptions = [
+            'capped' => true,
+            'max' => 100,
+            'size' => 1048576,
+        ];
+
+        $operation = new CreateCollection($databaseName, $cappedCollectionName, $cappedCollectionOptions);
+        $operation->execute($this->getPrimaryServer());
+
+        // Insert documents into the capped collection.
+        $cappedNamespace = $databaseName . '.' . $cappedCollectionName;
+        $bulkWrite = new BulkWrite(['ordered' => true]);
+        $bulkWrite->insert(['_id' => 1]);
+        $result = $this->manager->executeBulkWrite($cappedNamespace, $bulkWrite);
+
+        // Get a cursor of type TAILABLE_AWAIT.
+        $operation = new Find($databaseName, $cappedCollectionName, [], ['cursorType' => 3, 'maxAwaitTimeMS' => $maxAwaitTimeMS]);
+        $cursor = $operation->execute($this->getPrimaryServer());
+        $it = new \IteratorIterator($cursor);
+
+        // Make sure we await results for no more than the maxAwaitTimeMS.
+        $it->rewind();
+        $it->next();
+        $startTime = microtime(true);
+        $it->next();
+        $this->assertLessThan($maxAwaitTimeMS, (microtime(true) - $startTime));
+
+        // Drop the capped collection.
+        $operation = new DropCollection($databaseName, $cappedCollectionName);
+        $operation->execute($this->getPrimaryServer());
```

Comment:
I noticed that on a test failure, attempting to re-run this test will immediately encounter an exception attempting to create the capped collection ("a collection 'phplib_test.FindFunctionalTest.f36e3691.capped' already exists").

If we use the default collection name from `$this->getCollectionName()`, the `setUp()` method from `FunctionalTestCase` will take care of dropping that collection for us before the test runs (ensuring a blank slate). Additionally, `tearDown()` will also drop it after a test completes. If the test fails, `tearDown()` leaves it in place, which is helpful in case you'd like to inspect its state manually (e.g. poking around with the mongo shell).

I realize that the MapReduce test cases I referenced that use a `$this->getCollectionName() . '.output'` collection only drop it at the end (assuming earlier assertions all succeed and the test makes it up to that point). The reason we don't have to drop the ".output" collection for those tests is that [`mapReduce`](https://docs.mongodb.com/manual/reference/command/mapReduce/) happily overwrites it (unlike the [`create`](https://docs.mongodb.com/manual/reference/command/create/) command).